baybe/utils/basic.py

```diff
@@ -181,3 +182,36 @@ def find_subclass(base: type, name_or_abbr: str, /):
             f"The class name or abbreviation '{name_or_abbr}' does not refer to any "
             f"of the subclasses of '{base.__name__}'."
         )
+
+
+def register_hook(target: Callable, hook: Callable) -> Callable:
+    """Register a hook for the given target method.
```

Comment:
technically it doesnt have to be a method, perhaps `function` or `callable` is more precise